codalab/apps/web/static/js/worksheet/table_bundle_interface.js

```diff
@@ -32,11 +32,28 @@ var TableBundle = React.createClass({
             else
               this.focusOnRow(newIndex);
         }.bind(this), 'keydown');
+
+        Mousetrap.bind(['enter'], function(e) {
+            window.open(this.refs['row' + this.state.rowFocusIndex].props.url, '_blank');
+        }.bind(this), 'keydown');
+
+        // Paste uuid of focused bundle into console
+        Mousetrap.bind(['u'], function(e) {
+            var uuid = this.refs['row' + this.state.rowFocusIndex].props.uuid;
+            $('#command_line').terminal().insert(uuid);
+        }.bind(this), 'keydown');
+
+        // Paste args of focused bundle into console
+        Mousetrap.bind(['a'], function(e) {
+            var bundleInfo = this.refs['row' + this.state.rowFocusIndex].props.bundleInfo;
+            if (bundleInfo.command != null) {
+                $('#command_line').terminal().insert(bundleInfo.command);
```

Comment:
We should stick it into bundle_info before shipping it off to the client to
avoid an extra server call.

On Tue, Nov 17, 2015 at 6:58 PM, Stephen Koo <notifications@github.com>
wrote:

> In codalab/apps/web/static/js/worksheet/table_bundle_interface.js
> <https://github.com/codalab/codalab/pull/1414#discussion_r45154409>:
>
> > +
> > +        Mousetrap.bind(['enter'], function(e) {
> > +            window.open(this.refs['row' + this.state.rowFocusIndex].props.url, '_blank');
> > +        }.bind(this), 'keydown');
> > +
> > +        // Paste uuid of focused bundle into console
> > +        Mousetrap.bind(['u'], function(e) {
> > +            var uuid = this.refs['row' + this.state.rowFocusIndex].props.uuid;
> > +            $('#command_line').terminal().insert(uuid);
> > +        }.bind(this), 'keydown');
> > +
> > +        // Paste args of focused bundle into console
> > +        Mousetrap.bind(['a'], function(e) {
> > +            var bundleInfo = this.refs['row' + this.state.rowFocusIndex].props.bundleInfo;
> > +            if (bundleInfo.command != null) {
> > +                $('#command_line').terminal().insert(bundleInfo.command);
>
> OK, through an AJAX call to the client? Or maybe some hack through the web
> terminal?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab/pull/1414/files#r45154409>.
>